src/VisualStudio/InteractiveServices/Interactive/VsInteractiveWindowPackage.cs

```diff
@@ -38,6 +39,12 @@ protected override void Initialize()
             var menuCommandService = (OleMenuCommandService)GetService(typeof(IMenuCommandService));
             InitializeMenuCommands(menuCommandService);
             InitializeResetInteractiveFromProjectCommand(menuCommandService);
+
+            var telemetrySetupExtensions = _componentModel.GetExtensions<IRoslynTelemetrySetup>();
```

Comment:
make sure this happens on UI thread. also can you change RoslynTelemetrySetup to actually derived from ThreadAffinitizedObject and check whether it is called from UI thread in initialize